### ISSUE TITLE
spells range should be standardized

### ISSUE THREAD

**Issue by TSPARR** [ZachMyers3/flask-dnd#5]

**Comment by ZachMyers3** [comment]
Edge cases that need addressed:  
- Touch
- Self
- Unlimited
- Self (5-foot radius.. move to description?)
- Sight
- Miles (convert to feet?)

Hoping to represent this field as an integer in feet, not sure how to address these cases.

**Comment by TSPARR** [comment]
Maybe we add this and #3 into a dict together? Where touch is a boolean and an instantaneous action has a duration of 0 and that way we can turn duration into an int as well I think. As well as pull from description the number of affected to account for self with range being an int. 
``` 
 {
    "name": "Misty Step",
    "desc": "Briefly surrounded by silvery mist, you teleport up to 30 feet to an unoccupied space that you can see.",
    "book": "Player's Handbook",
    "page": 260,
    "components": {
      "material": false,
      "materials_needed": "",
      "raw": "V",
      "somatic": false,
      "verbal": true
    },
    "casting": {
      "range": 0,
      "affected": "Self",
      "casting_time": 6,
      "action_type": "Bonus",
      "duration": 0,
      "concentration": false,
      "ritual": false,
      "touch": false
    },
    "level": 2,
    "school": "Conjuration",
    "classes": [
      "Druid",
      "Paladin",
      "Sorcerer",
      "Warlock",
      "Wizard"
    ],
    "archetype": "Druid: Coast<br/> Paladin: Ancients, Vengeance",
    "circles": "Coast",
    "oaths": "Ancients, Vengeance"
  }
```

```
{
    "name": "Power Word Heal",
    "desc": "A wave of healing energy washes over the creature you touch. The target regains all its hit points. If the creature is charmed, frightened, paralyzed, or stunned, the condition ends. If the creature is prone, it can use its reaction to stand up. This spell has no effect on undead or constructs.",
    "book": "Player's Handbook",
    "page": 266,
    "components": {
      "material": false,
      "materials_needed": "",
      "raw": "V, S",
      "somatic": true,
      "verbal": true
    },
    "casting": {
      "range": 0,
      "affected": "1",
      "casting_time": 6,
      "action_type": "Action",
      "duration": 0,
      "concentration": false,
      "ritual": false,
      "touch": true
    },
    "level": 9,
    "school": "Evocation",
    "classes": [
      "Bard"
    ]
  }
```


Side note: Archetype, Circles, Oaths, Pacts and a few other things need looking at eventually.

**Comment by ZachMyers3** [comment]
I think this is the best way to represent all of this, I'll work on a migration script for this similar to materials.

**Comment by ZachMyers3** [comment]
How would options within the spell description be handled here?

**Comment by ZachMyers3** [comment]
Weird edge case of special for duration:

```json
{
    "name": "Creation",
    "desc": "You pull wisps of shadow material from the Shadowfell to create a nonliving object of vegetable matter within 'range': soft goods, rope, wood, or something similar. You can also use this spell to create mineral objects such as stone, crystal, or metal. The object created must be no larger than a 5-foot cube, and the object must be of a form and material that you have seen before.The duration depends on the object’s material. If the object is composed of multiple materials, use the shortest duration.<b>Vegetable matter</b> 1 day<b>Stone or crystal</b> 12 hours<b>Precious metals</b> 1 hour<b>Gems</b> 10 minutes<b>Adamantine or mithral</b> 1 minuteUsing any material created by this spell as another spell’s material component causes that spell to fail.",
    "higher_level": "When you cast this spell using a spell slot of 6th level or higher, the cube increases by 5 feet for each slot level above 5th.",
    "book": "Player's Handbook",
    "page": 229,
    "range": "30 feet",
    "components": {
      "material": true,
      "materials_needed": "A tiny piece of matter of the same type of the item you plan to create.",
      "raw": "V, S, M",
      "somatic": true,
      "verbal": true
    },
    "ritual": false,
    "duration": "Special",
    "concentration": false,
    "casting_time": "1 minute",
    "level": 5,
    "school": "Illusion",
    "classes": [
      "Sorcerer",
      "Wizard"
    ]
  }
```

**Comment by ZachMyers3** [comment]
Was able to split by the '/' and create an array, then convert the distinct times that way. How the new schema looks:

```json
 {
    "name": "Misty Step",
    "desc": "Briefly surrounded by silvery mist, you teleport up to 30 feet to an unoccupied space that you can see.",
    "book": "Player's Handbook",
    "page": 260,
    "components": {
      "material": false,
      "materials_needed": "",
      "raw": "V",
      "somatic": false,
      "verbal": true
    },
    "casting": {
      "range": 0,
      "affected": "Self",
      "casting_time": [
          6
      ]
      "action_type": "Bonus",
      "duration": 0,
      "concentration": false,
      "ritual": false,
      "touch": false
    },
    "level": 2,
    "school": "Conjuration",
    "classes": [
      "Druid",
      "Paladin",
      "Sorcerer",
      "Warlock",
      "Wizard"
    ],
    "archetype": "Druid: Coast<br/> Paladin: Ancients, Vengeance",
    "circles": "Coast",
    "oaths": "Ancients, Vengeance"
  }
```

**Comment by ZachMyers3** [comment]
Fixed in fc7300c36b02e0c121a0c133df729b2a9ce31cbd
